src/EnergyPlus/DataSurfaceColors.cc

```diff
@@ -79,15 +79,34 @@ bool MatchAndSetColorTextString(EnergyPlusData &state,
     //       MODIFIED       na
     //       RE-ENGINEERED  na
 
-    bool WasSet = false;
-    int found = UtilityRoutines::FindItem(String, state.dataSurfColor->colorkeys, static_cast<int>(ColorNo::NUM));
-    if (found != 0) {
-        if (ColorType == "DXF") {
-            state.dataSurfColor->DXFcolorno(found) = SetValue;
-            WasSet = true;
-        }
-    }
-    return WasSet;
+    static constexpr std::array<std::string_view, static_cast<int>(DataSurfaceColors::ColorNo::NUM)> colorkeys = {
+        "TEXT",
+        "WALLS",
+        "WINDOWS",
+        "GLASSDOORS",
+        "DOORS",
+        "ROOFS",
+        "FLOORS",
+        "DETACHEDBUILDINGSHADES",
+        "DETACHEDFIXEDSHADES",
+        "ATTACHEDBUILDINGSHADES",
+        "PHOTOVOLTAICS",
+        "TUBULARDAYLIGHTDOMES",
+        "TUBULARDAYLIGHTDIFFUSERS",
+        "DAYLIGHTREFERENCEPOINT1",
+        "DAYLIGHTREFERENCEPOINT2",
+    };
+
+    // ColorType must be "DXF"
+    if (ColorType != "DXF") return false;
+
+    // try to find enum value
+    int foundIdx = getEnumerationValue(colorkeys, UtilityRoutines::MakeUPPERCase(String));
+    if (foundIdx == -1) return false;
+
+    // if we've made it here, we found the value
+    state.dataSurfColor->DXFcolorno[foundIdx] = SetValue;
+    return true;
```

Comment:
I refactored this function because I'm pretty sure it was broken before. `FindItem` uses a 1-based lookup, but these Array1Ds were definitely 0-based before. This and the new unit test could use an extra 5 seconds from reviewers.